node/network/statement-distribution/src/lib.rs

```diff
@@ -1751,45 +1790,12 @@ impl StatementDistribution {
 						"New active leaf",
 					);
 
-					let (validators, session_index) = {
-						let (val_tx, val_rx) = oneshot::channel();
-						let (session_tx, session_rx) = oneshot::channel();
-
-						let val_message = AllMessages::RuntimeApi(
-							RuntimeApiMessage::Request(
-								relay_parent,
-								RuntimeApiRequest::Validators(val_tx),
-							),
-						);
-						let session_message = AllMessages::RuntimeApi(
-							RuntimeApiMessage::Request(
-								relay_parent,
-								RuntimeApiRequest::SessionIndexForChild(session_tx),
-							),
-						);
-						ctx.send_messages(
-							std::iter::once(val_message).chain(std::iter::once(session_message))
-						).await;
-
-						match (val_rx.await?, session_rx.await?) {
-							(Ok(v), Ok(s)) => (v, s),
-							(Err(e), _) | (_, Err(e)) => {
-								tracing::warn!(
-									target: LOG_TARGET,
-									err = ?e,
-									"Failed to fetch runtime API data for active leaf",
-								);
-
-								// Lacking this bookkeeping might make us behave funny, although
-								// not in any slashable way. But we shouldn't take down the node
-								// on what are likely spurious runtime API errors.
-								return Ok(false)
-							}
-						}
-					};
+					let session_index = runtime.get_session_index(ctx, relay_parent).await?;
+					let info = runtime.get_session_info_by_index(ctx, relay_parent, session_index).await?;
+					let session_info = &info.session_info;
```

Comment:
It's not clear to me where the `&mut authorities` is actually written into. I thought it'd be somewhere around here, but I only see the place where it's read from below.

Reply:
It is written on network update (`PeerConnected` events).

Reply:
(and disconnect of course).